### PR TITLE
Add UKHSA ONS theme

### PR DESCRIPTION
**Pull request by katie-gardner**
### What is the context of this PR?
This PR adds the UKHSA ONS theme.

[Validator PR ](https://github.com/ONSdigital/eq-questionnaire-validator/pull/220) for allowing the theme

### How to review

- Use `test_theme_ukhsa_ons.json` and check that the logos at the top show correctly. 
- Check that implementation is similar to that of other themes.
- Check that the logos stack vertically when the screen size is reduced

### Checklist
* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Comment by katie-gardner** [comment]
[New theme for UKHSA Branding (WCIS) (1 SP)](https://trello.com/c/6axFtNLI/5913-new-theme-for-ukhsa-branding-wcis-1-sp)

File: scripts/run_validator.sh [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env bash
-tag=latest
+tag=add-ukhsa-ons-theme
```
Comment (katie-gardner) [inline scripts/run_validator.sh:2]: reminder to revert

**Review by rmccar — CHANGES_REQUESTED** [review]
The logos dont quite line up. I think we should either make the ONS one a bit bigger or add the `ons-u-mb-xs` class to it

Screenshot 2023-10-09 at 16 55 15

Other than that looks good

File: schemas/test/en/test_theme_ukhsa_ons.json [hunk 2]
```diff
@@ -0,0 +1,75 @@
+{
+    "mime_type": "application/json/ons/eq",
+    "language": "en",
+    "schema_version": "0.0.1",
+    "data_version": "0.0.3",
+    "survey_id": "0",
+    "title": "Test Winter Coronavirus Infection Study",
```
Comment (MebinAbraham) [inline schemas/test/en/test_theme_ukhsa_ons.json:7]: I would remove this from the title. Just name it after UKHSA. The survey isn't important, they may run many.

File: templates/errors/_base.html [hunk 3]
```diff
@@ -2,6 +2,6 @@
 
 {% set SURVEY_TYPES_BUSINESS = ["northernireland", "business", "dbt", "dbt-ni", "dbt-dsit", "dbt-dsit-ni", "orr", "desnz", "desnz-ni"] %}
 {% set SURVEY_TYPES_DEFAULT = ["default"] %}
-{% set SURVEY_TYPES_SOCIAL = ["social"] %}
+{% set SURVEY_TYPES_SOCIAL = ["social", "ukhsa-ons"] %}
 {% set SURVEY_TYPES_HEALTH = ["health"] %}
```
[the remark this answers is not in the document]
Comment (katie-gardner) [reply templates/errors/_base.html:6]: Ah thanks, fixed

**Comment by katie-gardner** [comment]
> The logos dont quite line up. I think we should either make the ONS one a bit bigger or add the `ons-u-mb-xs` class to it
> 
> Screenshot 2023-10-09 at 16 55 15
> 
> Other than that looks good

Thanks Rich - have tried adding the margin class you suggested, let me know what you think?

**Comment by rmccar** [comment]
> > The logos dont quite line up. I think we should either make the ONS one a bit bigger or add the `ons-u-mb-xs` class to it
> > Screenshot 2023-10-09 at 16 55 15
> > Other than that looks good
> 
> Thanks Rich - have tried adding the margin class you suggested, let me know what you think?

Looks much better 👍

**Review by rmccar — APPROVED** [review]

**Comment by MebinAbraham** [comment]
> > > The logos dont quite line up. I think we should either make the ONS one a bit bigger or add the `ons-u-mb-xs` class to it
> > > Screenshot 2023-10-09 at 16 55 15
> > > Other than that looks good
> > 
> > 
> > Thanks Rich - have tried adding the margin class you suggested, let me know what you think?
> 
> Looks much better 👍

The logo sizing is currently inconsistent. Is the header sizing dependent on the max width of the logos @rmccar. There doesn't seem to be a min height?.

This compared to `desnz_ni` is smaller but matches the header size of dbt_dsit_ni. The headers should all be the same height ideally. 

Should they be aligned vertically?. This is what I can see at the moment. But this isn't centered among themselves.
image

**Comment by rmccar** [comment]
> > > > The logos dont quite line up. I think we should either make the ONS one a bit bigger or add the `ons-u-mb-xs` class to it
> > > > Screenshot 2023-10-09 at 16 55 15
> > > > Other than that looks good
> > > 
> > > 
> > > Thanks Rich - have tried adding the margin class you suggested, let me know what you think?
> > 
> > 
> > Looks much better 👍
> 
> The logo sizing is currently inconsistent. Is the header sizing dependent on the max width of the logos @rmccar. There doesn't seem to be a min height?.
> 
> This compared to `desnz_ni` is smaller but matches the header size of dbt_dsit_ni. The headers should all be the same height ideally.
> 
> Should they be aligned vertically?. This is what I can see at the moment. But this isn't centred among themselves. image

When you say header size do you mean the height of the white part of the header? Because yes thats based on the hight of the logos. There is a minimum though.

With so many different logos created in different ways its fiddly to get them all the right size. Some have really different designs too so need to be a certain size to be readable. Plus we have stacked and non stacked versions of some logos like the ONS one. There seems to be no standardisation when creating them and when Ive done these in the past Ive had to tidy up and get the logos looking a similar size myself by hand and I don't have any software to do it. I helped Katie out with this yesterday because the logo attached to the card doesn't work at all.

I think really we need a designer to go over all the logos to standardise and optimise them.

On those logos being aligned that was the best way of doing it with the DS utility classes we have. Another option would be to potentially edit the ONS logo to make the whitespace around it a bit bigger in order to pad it out. I can have a look at that if you wanted?

File: app/survey_config/social_survey_config.py [hunk 4]
```diff
@@ -63,3 +63,13 @@ def get_footer_legal_links(self, cookie_has_theme: bool) -> Optional[list[dict]]
...
+
+
+@dataclass
+class UKHSAONSSocialSurveyConfig(SocialSurveyConfig):
+    masthead_logo: str = read_file(
+        "./templates/assets/images/ons-logo-stacked.svg"
+    ) + read_file("./templates/assets/images/ukhsa-logo-stacked.svg")
+    masthead_logo_mobile: str = read_file(
+        "./templates/assets/images/ons-logo-stacked.svg"
+    ) + read_file("./templates/assets/images/ukhsa-logo-stacked.svg")
```
[the remark this answers is not in the document]
Comment (MebinAbraham) [reply app/survey_config/social_survey_config.py:75]: Got confirmation, the UKHSA logo needs to go first.

**Comment by MebinAbraham** [comment]
> When you say header size do you mean the height of the white part of the header? Because yes thats based on the hight of the logos. There is a minimum though.

How is that currently calculated?. Is it the largest logo etc?. At the moment, it doesn't seem consistent. 

For now, we can get it in as is. Can you raise a card to review these from a design perspective. Is there any change needed in DS?.

**Comment by rmccar** [comment]
> > When you say header size do you mean the height of the white part of the header? Because yes thats based on the hight of the logos. There is a minimum though.
> 
> How is that currently calculated?. Is it the largest logo etc?. At the moment, it doesn't seem consistent.
> 
> For now, we can get it in as is. Can you raise a card to review these from a design perspective. Is there any change needed in DS?.

Yeah itll be the largest one that its based on.

Yeah Ill raise a card and add it to the backlog. No changes in the DS needed as all the logos we use are in runner

**Pull request merged**